cleverhans/utils.py

```diff
@@ -220,6 +220,95 @@ def grid_visual(data):
     return figure
 
 
+def get_logits_over_interval(sess, model, x, adv_x, x_data,
+                             min_epsilon=-10, max_epsilon=10,
+                             num_points=21):
+    """Get logits when the input is perturbed in an interval in adv direction.
+
+    Args:
+        sess: Tf session
+        model: Model for which we wish to get logits
+        x: Tf placeholder for data
+        adv_x_real: Tf placeholder for adv perturbation of x
+        x_data: Numpy array corresponding to the data slice used to generate plot
+        min_epsilon: Minimum value of epsilon over the interval
+        max_epsilon: Maximum value of epsilon over the interval
+        num_points: Number of points used to interpolate
+
+    Returns:
+        Numpy array containing log probabilities
+
+    Raises:
+        ValueError if min_epsilon is larger than max_epsilon
+    """
+    if min_epsilon > max_epsilon:
+        raise ValueError('Minimum epsilon is less than maximum epsilon')
+
+    delta = (max_epsilon - min_epsilon) / (num_points - 1)
+    epsilon = min_epsilon
+    eta = adv_x - x
```

Comment:
eta needs to be normalized in some way, otherwise you multiply the epsilon that was used inside the attack by the epsilon used in your for loop below.
For example, if `adv_x` came from an FGSM attack with eps=.3 and `max_epsilon` is 3, the largest adversarial perturbation we'd get in the for loop would be size .09

Reply:
I had a tf.nn.l2_normalize on eta, but took it out since Nicolas said importing tensorflow at the top level caused the build to fail. I imported it within the function def.